apps/passport-server/src/services/generic-issuance/pipelines/PretixPipeline.ts

```diff
@@ -883,19 +900,27 @@ export class PretixPipeline implements BasePipeline {
       const now = new Date();
 
       if ((this.definition.options.semaphoreGroups ?? []).length > 0) {
-        // TODO check the timeUpdated returned from save(), which will equal
-        // now if a new or updated commitment was recorded, in which case the
-        // Semaphore groups need to be updated. They will be updated on next
-        // load(), but this means there's some latency between a new user
-        // authenticating and being added to their Semaphore groups.
-
-        // Consumer is validated, so save them in the consumer list
-        await this.consumerDB.save(
+        const consumer = await this.consumerDB.save(
           this.id,
           emailPCD.claim.emailAddress,
           emailPCD.claim.semaphoreId,
           now
         );
+
+        // If the update time is now, that means our save() operation caused
+        // a change in the consumer DB, and we have to update the Semaphore
+        // groups. This is rare, since most saves will not cause a change.
+        // It will, however, always occur the first time the user hits a
+        // feed with Semaphore groups enabled (which might not be the first
+        // time they *ever* hit that feed, if Semaphore groups were not
+        // initialy enabled).
+        // It seems reasonable to await on the update here, even though it will
+        // slow down the response time, as it ensures that the user is present
+        // in all of the necessary Semaphore groups.
+        if (consumer.timeUpdated.getTime() === now.getTime()) {
```

Comment:
if you made `this.consumerDB.save` return a boolean representing whether the semaphore group should be updated, you could save yourself from having to duplicate this comment.

Reply:
Oh, that would be simpler, yes.